core/src/test/kotlin/ch/ergon/dope/buildTest/QueryBuilderTest.kt

```diff
@@ -787,4 +792,21 @@ class QueryBuilderTest {
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support selecting case`() {
+        val someBucket = someBucket()
+        val expected = "SELECT CASE `numberField` WHEN `other` THEN `numberField` END, " +
+            "CASE WHEN `booleanField` THEN `numberField` ELSE `stringField` END AS `alias` FROM `someBucket`"
+
+        val actual = create
+            .select(
+                case(someNumberField()).whenThen(someNumberField("other").resultsIn(someNumberField())),
```

Comment:
I don't really like _whenThen_ with just one argument, the function-name implies two arguments, what about just `.condition()`?

Reply:
It generates WHEN...THEN in sql++ yes, but it might be confusing if you are working with MongoDB (or any other database)

Reply:
I renamed it to condition